### DISCUSSION TITLE
How to get 'HasUniqueRoleAssignments' Property from Web

### DISCUSSION THREAD

**Discussion by bajce** [pnp/pnpcore#575] [Q&A] [answered]
I'm trying to read all Webs and Lists from SharePoint Online SiteCollection and I also need to read `HasUniqueRoleAssignments` Property from Web. List PnP Core SDK Object have Property for `HasUniqueRoleAssignments` but Web doesn't, even do there is one if you use SharePoint PnP PowerShell Module or CSOM.

As a workaround im creating CSOM Context from existing PnP Core Context and then im loading that Property:

      using (var csomContext = PnPCoreSdk.Instance.GetClientContext(pnpCoreSdkContext))
      {
            csomContext.Load(csomContext.Web, web => web.HasUniqueRoleAssignments);
            csomContext.ExecuteQuery();

            csomContext.Web.HasUniqueRoleAssignments;
      }

This costs me performance (as I need to create new CSOM context and load the Web Property) and in to create CSOM Context from PnP Core SDK Context I need PnP Framework Nuget Package, which increases the project dependencies.

Is there any another way how to get `HasUniqueRoleAssignments` Property by only using PnP Core SDK **without use of any other Libraries**?

(StackOverflow [question](https://stackoverflow.com/questions/69492624/sharepoint-online-pnp-core-sdk-c-sharp-how-to-get-hasuniqueroleassignments-p))

**Comment by jansenbe** [comment, marked as answer]
@bajce : the `HasUniqueRoleAssignments` property has been added (see https://github.com/pnp/pnpcore/commit/bfb72c535962914762518a837935cda07a6e17d0). The next nightly build will have it (version 1.4.13 or higher). 

Thanks for using PnP Core SDK and providing the feedback, this will help improve this library 💪🚀

**Reply by bajce** [reply]
That's great! Thank you for the quick response.

**Reply by bajce** [reply]
Is it somehow possible to get `HasUniqueRoleAssignments` Property of List Item in version 1.4? If not is it planned for some of the future versions?

**Reply by jansenbe** [reply]
I can add that as well, then we have all the needed props on `IWeb`, `IList` and `IListItem`

**Reply by bajce** [reply]
That would be great. Thank You in advance!

**Reply by jansenbe** [reply]
@bajce : done, see https://github.com/pnp/pnpcore/commit/b37346c97ddf7c89c6d80b949831d5bc65cc9014
